app/actions/local/category.test.ts

```diff
@@ -66,4 +71,172 @@ describe('handleConvertedGMCategories', () => {
         expect(models).toBeDefined();
         expect(models!.length).toBe(3); // two for removing channel for a custom and a DM category, and one for adding it to default channels category
     });
+
+    it('error - no channel category', async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+
+        const {models, error} = await handleConvertedGMCategories(serverUrl, channelId, teamId1, true);
+        expect(error).toBeDefined();
+        expect(error).toBe('Failed to find default category when handling category of converted GM');
+        expect(models).toBeUndefined();
+    });
+
+    it('error - database not prepared', async () => {
+        const {error} = await handleConvertedGMCategories(serverUrl, channelId, teamId1, true);
+        expect(error).toBeDefined();
+    });
+});
+
+describe('category crud', () => {
+    const serverUrl = 'baseHandler.test.com';
+    const teamId1 = 'team_id_1';
+    const channelId1 = 'channel_id_1';
+    const channelId2 = 'channel_id_2';
+    const team: Team = {
+        id: teamId1,
+    } as Team;
+
+    let operator: ServerDataOperator;
+
+    beforeEach(async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+        await operator.handleTeam({teams: [team], prepareRecordsOnly: false});
+    });
```

Comment:
any reason to run this before each test instead of just once?

Reply:
Might be misunderstanding what's happening under the covers but I presumed it was giving a clean slate database to work off of for each test. I'd like to start clean to make sure the tests don't interact with each other. Is that not the case?

Reply:
To accomplish that you would need to delete the db after each.. if not it will re-use the existing db

Reply:
Got it. In that case I'll change this to just running once